test/app/components/market/submit/parameter-input.component.spec.ts

```diff
@@ -40,21 +51,234 @@ fdescribe("ParameterInputComponent", () => {
     let testComponent: TestComponent;
     let component: ParameterInputComponent;
     let de: DebugElement;
+    let storageServiceSpy: any;
+    let dialogServiceSpy: any;
+    let listProxy: RxMockListProxy<any, BatchApplication>;
 
     beforeEach(() => {
+        listProxy = new RxMockListProxy(BatchApplication, {
+            cacheKey: "id",
+            items: [
+                Fixtures.application.create({ id: "app-1" }),
+            ],
+        });
+
+        dialogServiceSpy = {
+            list: () => listProxy,
+            onApplicationAdded: new Subject(),
+            listContainers: () => listProxy,
+        };
+
+        storageServiceSpy = {
+            list: () => listProxy,
+            onApplicationAdded: new Subject(),
+            listContainers: () => listProxy,
+        };
+
         TestBed.configureTestingModule({
-            imports: [],
-            declarations: [ParameterInputComponent, TestComponent],
+            imports: [RouterTestingModule, ReactiveFormsModule, FormsModule, MaterialModule, NoopAnimationsModule],
+            declarations: [NoItemMockComponent, ParameterInputComponent,
+                TestComponent, FileGroupPickerComponent, CloudFilePickerComponent],
+            providers: [
+                { provide: StorageService, useValue: storageServiceSpy },
+                { provide: DialogService, useValue: dialogServiceSpy },
+            ],
             schemas: [NO_ERRORS_SCHEMA],
         });
+
         fixture = TestBed.createComponent(TestComponent);
         testComponent = fixture.componentInstance;
         de = fixture.debugElement.query(By.css("bl-parameter-input"));
         component = de.componentInstance;
         fixture.detectChanges();
     });
 
-    describe("1st tests", () => {
-        it("true is true", () => expect(true).toBe(true));
+    it("should propogate form control value to class", () => {
+        expect(testComponent.paramControl.value).toBe(component.parameterValue.value);
+    });
+
+    describe("text parameter type", () => {
+        let inputEl: DebugElement;
+        const initialInput: string = "initial input value";
```

Comment:
don't define the type here is is a bit too verbose. Only do the type if it is not obious or on a class attribute

Reply:
removed types for simple variables